packages/hebao_v1/contracts/version1.1/security/GuardianModule.sol

```diff
@@ -0,0 +1,206 @@
+        )
+        external
+        nonReentrant
+        onlyWhenWalletUnlocked(wallet)
+        notWalletOwner(wallet, guardian)
+        onlyFromWallet(wallet)
+    {
+        require(guardian != address(0), "ZERO_ADDRESS");
+        require(group < GuardianUtils.MAX_NUM_GROUPS, "INVALID_GROUP");
+        uint numGuardians = controller.securityStore().numGuardiansWithPending(wallet);
+        require(numGuardians < MAX_GUARDIANS, "TOO_MANY_GUARDIANS");
+
+        uint effectiveTime = now;
+        if (numGuardians >= MIN_ACTIVE_GUARDIANS) {
+            effectiveTime = now + pendingPeriod;
+        }
+        controller.securityStore().addGuardian(wallet, guardian, group, effectiveTime);
+        emit GuardianAdded(wallet, guardian, group, effectiveTime);
+    }
+
+    function cancelGuardianAddition(
+        address wallet,
+        address guardian
+        )
+        external
+        nonReentrant
+        onlyWhenWalletUnlocked(wallet)
+        onlyFromWallet(wallet)
+    {
+        controller.securityStore().cancelGuardianAddition(wallet, guardian);
+        emit GuardianAdditionCancelled(wallet, guardian);
+    }
+
+    function removeGuardian(
+        address wallet,
+        address guardian
+        )
+        external
+        nonReentrant
+        onlyWhenWalletUnlocked(wallet)
+        onlyWalletGuardian(wallet, guardian)
+        onlyFromWallet(wallet)
+    {
+        controller.securityStore().removeGuardian(wallet, guardian, now + pendingPeriod);
+        emit GuardianRemoved(wallet, guardian, now + pendingPeriod);
+    }
+
+    function cancelGuardianRemoval(
+        address wallet,
+        address guardian
+        )
+        external
+        nonReentrant
+        onlyWhenWalletUnlocked(wallet)
+        onlyFromWallet(wallet)
+    {
+        controller.securityStore().cancelGuardianRemoval(wallet, guardian);
+        emit GuardianRemovalCancelled(wallet, guardian);
+    }
+
+    function lock(address wallet)
+        external
+        nonReentrant
+        // onlyWhenWalletUnlocked(wallet)
+        onlyFromGuardian(wallet)
+        onlyHaveEnoughGuardians(wallet)
+    {
+        lockWallet(wallet);
+    }
+
+    function unlock(address wallet)
+        external
+        nonReentrant
+        // onlyWhenWalletLocked(wallet)
+        onlyFromGuardian(wallet)
+    {
+        unlockWallet(wallet, false);
+    }
+
+    /// @dev Recover a wallet by setting a new owner.
+    /// @param request The general request object.
+    /// @param newOwner The new owner address to set.
+    ///        The addresses must be sorted ascendently.
+    function recover(
+        SignedRequest.Request calldata request,
+        address newOwner
+        )
+        external
+        nonReentrant
+        notWalletOwner(request.wallet, newOwner)
+        onlyHaveEnoughGuardians(request.wallet)
+    {
+
+        Wallet w = Wallet(request.wallet);
+        address oldOwner = w.owner();
+        require(newOwner != oldOwner, "SAME_ADDRESS");
```

Comment:
the same as notWalletOwner(request.wallet, newOwner)

Reply:
Good point!